tests/client/core/TestAddonManager.js

```diff
@@ -91,12 +91,20 @@ describe('addonManager', () => {
     it('passes the installObj, the event and the id to the callback', () => {
       const fakeEvent = { type: 'fakeEvent' };
       return addonManager.install(
-        fakeInstallUrl, fakeCallback, { _mozAddonManager: fakeMozAddonManager })
+        fakeInstallUrl, fakeCallback, { _mozAddonManager: fakeMozAddonManager, src: 'home' })
         .then(() => {
           fakeInstallObj.onDownloadProgressListener(fakeEvent);
           assert.ok(fakeCallback.calledWith(fakeInstallObj, fakeEvent));
         });
     });
+
+    it('requires a src', () => (
+      addonManager.install(
+        fakeInstallUrl, fakeCallback, { _mozAddonManager: fakeMozAddonManager })
+          .then(
```

Comment:
Yeah I wish these aspects of Promises were more obvious - I always forget the precise details of the subtle differences.